python/GafferUI/MultiLineTextWidget.py

```diff
@@ -440,6 +462,56 @@ def getFixedLineHeight( self ) :
 
 		return self.__fixedLineHeight
 
+	def setLineNumbersVisible( self, visible ) :
+
+		self.__lineNumbersVisible = visible
+		self.__lineNumberAreaWidget.setVisible( visible )
+		self.__updateLineNumberAreaWidth( 0 )
+
+	def getLineNumbersVisible( self ) :
+
+		return self.__lineNumbersVisible
+
+	def lineNumberAreaWidth( self ) :
+
+		if not self.getLineNumbersVisible() :
+			return 0
+
+		digits = 1
+		value = max( 1, self.blockCount() )
+		while value >= 10 :
+			value *= 0.1
+			digits += 1
+
+		return 3 + self.fontMetrics().horizontalAdvance( '9' ) * digits
+
+	def lineNumberAreaPaintEvent( self, event ) :
+
+		block = self.firstVisibleBlock()
```

Comment:
In https://github.com/GafferHQ/gaffer/pull/6096/commits/a5e663177fa4065c04c07b8ccf2ae3f1d498ccf2 I changed the `Shift` + `Return` behavior to add a new block rather than just a new line. I think that's desirable? I don't know if anyone's using `Shift` + `Return` for some reason, but if so, this should be transparent to them and format better for us.